src/specs/datasource.test.js

```diff
@@ -135,4 +148,46 @@ describe('Scalyr datasource tests', () => {
     });
   });
 
+  describe('Annotation queries', () => {
+    let results;
+    beforeEach(() => {
+      results = [
+      {
+        timefield: 12345,
+        messagefield: "testmessage1",
+        timeendfield: 54321
+      },
+      {
+        timefield: 12345,
+        messagefield: "testmessage2",
+        timeendfield: 54321
+      },
+      {
+        timefield: 12345,
+        messagefield: "testmessage3",
+        timeendfield: 54321
+      }
+      ];
+    });
+    it('Should create a query request', () => {
+      const request = datasource.createLogsQueryForAnnotation(annotationQueryOptions, variables);
+      expect(request.url).toBe('proxied/query');
+      expect(request.method).toBe('POST');
+      const requestBody = JSON.parse(request.data);
+      expect(requestBody.token).toBe('123');
+      expect(requestBody.queryType).toBe('log');
+      expect(requestBody.startTime).toBe(sixHoursAgo.toISOString());
+      expect(requestBody.endTime).toBe(now.toISOString());
+    });
+
+    it('Should transform standard query results to annotations', () => {
```

Comment:
It would be good to also add test cases for all the edge cases in ``transformAnnotationResults`` function.